### ISSUE TITLE
Icinga2 installation on Windows (.NET framework)

### ISSUE THREAD

**Issue by dolefi** [Icinga/icinga2#7413]
## Is your feature request related to a problem? Please describe.
Hello, I have had time to look trough the documentation regarding upgrading windows clients to version 2.11+ 
What I do not understand is why the new instalator for windows client needs .NET framework version 4.6. We have large enviroment of servers with different versions of .NET frameworks that the applications running on those servers are dependant on and therefore we are simply unable to upgrade .NET framework sorely for Icinga installator (server2012 mainly).

![image](https://user-images.githubusercontent.com/17765008/62866132-626dcc00-bd10-11e9-8838-1411f01b8931.png)

## Describe the solution you'd like
Would it be possible, to create installator, without the need for .NET 4.6 if it is not application related dependency ? Or create installator without the GUI all together, so that installation via msiexec is possible.

## Describe alternatives you've considered
I can think of 2 sollutions, either cut down the need for .NET version, or simply create only command line installator (Lite version) if it is possible.

## Additional context
Thank you for all your inputs, please let me know, if this is something completely unrelated, and therefore cannot be done, or if it would be possible, as with the dependency on .NET 4.6 we are simply unable to even consider the 2.11 version of icinga. 

Best regards, 
Filip

**Comment by dnsmichi** [comment]
Hi,

this has been discussed for quite a while, and resulted in the evaluation and extensive tests in #7090. Previously we had to require users to install .NET 2.0 resp. 3.5 as additional package on Server 2016 and onwards. In order to resolve this requirement, we made the decision to require .NET 4.6 which is available on Windows 10 & Server 2016 or newer by default. This also enables us to improve the setup wizard in future versions, likewise adding support for ECC certificates in #7323. 

Older systems such as 2012 whose primary support already ended, need to install the .NET framework. Since Microsoft decided to drop support for specific older .NET versions, Windows update will bring the .NET 4.6 package to Windows hosts either way. And likely you should update them for security and performance reasons.

Our Windows package works as complete setup, providing both the binaries and the setup wizard. Users who prefer to automate their installation - like you mentioning msiexec - need to take care of the requirements beforehand.

There's no plans to split this up in the near future, as it complicates the setup experience which should be as convenient as possible. This is also the reason why the GUI exists on Windows, since not that many users prefer using the command line there.

TL;DR - with 2.11, .NET 4.6 is a definite requirement dropping the .NET 3.5 requirement. If you're planning to upgrade, evaluate whether this update conforms with your applications. I'd say if they are on 4.x already, there shouldn't be a problem with it.

Cheers,
Michael

**Comment by elderorange** [comment]
I totally agree to your argumentation. 
We have deadlines for these systems running older .net versions.
Till then I found a way to let them run on icinga2 version 2.11.2.

Can a post the workarounds that I found?

**Comment by dolefi** [comment]
Hi, well in the end I forcefully removed .NET check from the package itself (.msi).
That lets us install the client completely without any issues and as long as we don't use the graphical binary everything works as it should.

**Comment by elderorange** [comment]
Jap, that's the cleaner way by dropping the requierment.
Another dirty one ist to overwrite an existing version by an already installed client folder skipping the relevant configs and certs.

**Comment by dnsmichi** [comment]
Sure, why not. Keep in mind that you won't get any support for these modified versions though.
